### PR TITLE
add metarange ID to commit log and show

### PR DESCRIPTION
**Pull request by ozkatz**
Should aid in debugging user issues

### CODE REVIEW

**Comment by ozkatz** [comment]
This will come in handy with issues like #1263 (or any other hard to explain behavior where we might want metadata from the user in order to reproduce)

File: catalog/model.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type CommitLog struct {
 	Message      string    `db:"message"`
 	CreationDate time.Time `db:"creation_date"`
 	Metadata     Metadata  `db:"metadata"`
+	MetaRangeID  string    `db:"meta_range_id"`
```
Comment (itaiad200) [inline catalog/model.go:40]: DDL upgrade/downgrade needed?

File: api/api_controller.go [hunk 2]
```diff
@@ -411,6 +411,7 @@ func (c *Controller) GetCommitHandler() commits.GetCommitHandler {
 			Message:      commit.Message,
 			Metadata:     commit.Metadata,
 			Parents:      commit.Parents,
+			MetarangeID:  commit.MetaRangeID,
```
Comment (itaiad200) [inline api/api_controller.go:414]: nit: let's stick with `Metarange` or `MetaRange`.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Blocking since I believe DB upgrade is needed. Other than that LGTM.

File: api/api_controller.go (hunk 2 above)
Comment (ozkatz) [reply api/api_controller.go]: Fixed

File: catalog/model.go (hunk 1 above)
Comment (ozkatz) [reply catalog/model.go:40]: No, meta_range_id column introduced in [DDL 20](https://github.com/treeverse/lakeFS/blob/master/ddl/000020_tree_to_range.up.sql#L3).

**Review by arielshaqed — APPROVED** [review]
Approving because it helps, but not a fan.

Metarange ID is an implementation detail.  It should be visible to plumbing, but this PR adds it to porcelain.  E.g. in `git` I think you need to take an extra step _on plumbing_ to get tree ID from commit ID, e.g. use the `%T` format (`git show --format='%T' ...`) or [tree object](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects#tree_objects) shows how to `git show` the tree of a commit using the `BRANCH^{tree}` syntax.

So I'd rather have it displayed only if a format option is specified.  We can start with individual format flags for now, and go for full formats... later.  WDYT?

**Review by itaiad200 — APPROVED** [review]

**Comment by ozkatz** [comment]
@arielshaqed I've added a `--show-meta-range-id` flag to `lakectl show` and `lakectl log`. By default the meta range ID will be hidden unless this flag is set.

File: cmd/lakectl/cmd/log.go [hunk 3]
```diff
@@ -44,14 +46,17 @@ var logCmd = &cobra.Command{
 		if err != nil {
 			DieErr(err)
 		}
+		showMetaRangeID, _ := cmd.Flags().GetBool("show-meta-range-id")
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/log.go:49]: Sure is better than `--format "%H"` (as the _only_ way I could find to convince git to show me a tree-ID)!

**Review by arielshaqed — APPROVED** [review]
Great, thanks!

**Pull request merged**